corehq/pillows/xform.py

```diff
@@ -157,6 +163,65 @@ def get_xform_to_elasticsearch_pillow(pillow_id='XFormToElasticsearchPillow', nu
     )
 
 
+def get_xform_pillow(pillow_id='xform-pillow', ucr_division=None,
+                     include_ucrs=None, exclude_ucrs=None,
+                     num_processes=1, process_num=0, ucr_configs=None, skip_ucr=False,
+                     processor_chunk_size=UCR_PROCESSING_CHUNK_SIZE, topics=None, **kwargs):
+    # avoid circular dependency
+    from corehq.pillows.reportxform import transform_xform_for_report_forms_index, report_xform_filter
+    from corehq.pillows.mappings.user_mapping import USER_INDEX
+    if topics:
+        assert set(topics).issubset(FORM_TOPICS), "This is a pillow to process cases only"
+    topics = topics or FORM_TOPICS
+    change_feed = KafkaChangeFeed(
+        topics, client_id=pillow_id, num_processes=num_processes, process_num=process_num
+    )
+
+    ucr_processor = ConfigurableReportPillowProcessor(
+        data_source_providers=[DynamicDataSourceProvider(), StaticDataSourceProvider()],
+        ucr_division=ucr_division,
+        include_ucrs=include_ucrs,
+        exclude_ucrs=exclude_ucrs,
+    )
+    xform_to_es_processor = ElasticProcessor(
+        elasticsearch=get_es_new(),
+        index_info=XFORM_INDEX_INFO,
+        doc_prep_fn=transform_xform_for_elasticsearch,
+        doc_filter_fn=xform_pillow_filter,
+    )
+    unknown_user_form_processor = UnknownUsersProcessor()
+    form_meta_processor = FormSubmissionMetadataTrackerProcessor()
+    checkpoint_id = "{}-{}-{}-{}".format(
+        pillow_id, XFORM_INDEX_INFO.index, REPORT_XFORM_INDEX_INFO.index, USER_INDEX)
```

Comment:
@sravfeyn I know this was a long time ago, but out of curiosity, why does this include the `USER_INDEX` in this checkpoint ID? Is there some dependency on the form processor with the users index? 

cc: @orangejenny

Reply:
presumably because of the `UnknownUsersProcessor` which updates the user index. Though I'm not sure that if we were going to reindex the user index that we'd want to have to go through all the form data to add in 'unknown' users.

Reply:
Yeah, it's because of the `UnknownUsersProcessor`. But yeah, for large domains it doesn't make sense